src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs

```diff
@@ -782,8 +782,11 @@ private async Task AddDocumentsWithPotentialConflictsAsync(IEnumerable<Document>
                         var stringAndCommentTextSpansInSingleSourceTree = renameLocations
                             .Where(l => l.DocumentId == documentId && l.IsRenameInStringOrComment)
                             .GroupBy(l => l.ContainingLocationForStringOrComment)
-                            .Select(g => g.Key)
-                            .ToSet();
+                            .ToImmutableDictionary(
+                                g => g.Key,
+                                g => g.All(l => g.Key.Contains(l.Location.SourceSpan))
+                                     ? g.Select(l => new TextSpan(l.Location.SourceSpan.Start - g.Key.Start, l.Location.SourceSpan.Length)).ToImmutableSortedSet()
+                                     : null);
```

Comment:
no.  no no non no :)   

just rewrite this to be clear.

Reply:
Done :)